### PR TITLE
Dependency handling

### PR DESCRIPTION
**Pull request by joakimmoller**
This solves some disturbing version conflicts i have seen when different lambdas have different requirements.

The '-c' allows us to compile constraints from the list of all packages needed in the total environment. From that constraints file each lambda environment can fix the versions of its own requirements.

Replaced the build_requirements.sh with a makefile.

### CODE REVIEW

**Comment by joakimmoller** [comment]
actually, the constraints.txt is the requirements-dev.txt. I'll fix!

File: makefile [hunk 1]
```diff
@@ -0,0 +1,36 @@
+## Manage python dependencies:
```
Comment (skymandr) [inline makefile:1]: Traditionally Makefiles are named with capital M. Not sure why.

File: makefile [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+## make update -- Recompute the .txt requirements files files from
+##                scratch, updating all packages unless pinned in the
+##                .in files.
+
+
+DEVELOPER_ENV := requirements-dev.in
+
+PIP_COMPILE := pip-compile -q --no-header --resolver=backtracking
+CONSTRAINTS_ENV := $(addsuffix .txt, $(basename $(DEVELOPER_ENV)))
+MKFILE_PATH := $(abspath $(lastword $(MAKEFILE_LIST)))
```
Comment (skymandr) [inline makefile:16]: Where does `MAKEFILE_LIST` come from?

File: requirements-dev.in [hunk 3]
```diff
@@ -1,3 +1,5 @@
+-c ${CONSTRAINTS}
+-r requirements.in
```
Comment (skymandr) [inline requirements-dev.in:2]: how about addind `botocore-types` and `boto3-types` to dev? (or if they're called `types-*`, I can never remember...)

**Review by skymandr — APPROVED** [review]
Very neat! A few questions, but nothing blocking.

File: makefile (hunk 2 above)
Comment (joakimmoller) [reply makefile:16]: it's a makefile internal variable

File: makefile (hunk 1 above)
Comment (joakimmoller) [reply makefile:1]: Will change

File: requirements-dev.in (hunk 3 above)
Comment (joakimmoller) [reply requirements-dev.in:2]: they are all pulled in from boto3-stubs[essential, stepfunctions]

Comment (skymandr) [reply requirements-dev.in:2]: Cool. That sounds more convenient.

**Pull request merged**